Snowplow/Internal/Session/SPSession.m

```diff
@@ -74,15 +75,15 @@ - (instancetype)initWithForegroundTimeout:(NSInteger)foregroundTimeout andBackgr
         _isNewSession = YES;
         self.tracker = tracker;
         self.dataPersistence = [SPDataPersistence dataPersistenceForNamespace:tracker.trackerNamespace];
-        NSDictionary *storedSessionDict = self.dataPersistence.session;
-        
-        if (storedSessionDict) {
+        NSMutableDictionary *storedSessionDict = self.dataPersistence.session.mutableCopy;
+        _userId = [self retrieveUserIdWithSessionDict:storedSessionDict];
+        if (storedSessionDict && _userId) {
```

Comment:
Does this check make sense? `_userId` will never be nil – I think the `retrieveUserIdWithSessionDict` always assigns it. And if the `storedSessionDict` could be nil, then the line above should be enclosed in the condition too?

Reply:
true. `_userId` can't be nil. I added the check for extra safety but it probably confuses more than helping.
About `storedSessionDict` as nil: the `retrieveUserIdWithSessionDict` doesn't require it non-nil, so it's ok to have the argument nil. It can sound weird as we call a method (`sp_stringForKey`) on that argument but on ObjC a call to a nil object returns always nil (no crashes).

Reply:
That makes sense, didn't realize that you can call any method on nil objects in ObjC.